### DISCUSSION TITLE
Environment variable in velero deployment yaml

### DISCUSSION THREAD

**Discussion by VickyWinner** [vmware-tanzu/velero#5663] [Velero featured]
Greetings! Velero community. 
 I am looking for key/value in [values.yaml](https://github.com/vmware-tanzu/helm-charts/blob/main/charts/velero/values.yaml) helm example for installing and configuring velero.  Where do I create environment variables for velero deployment for passing client_id and secret_id for Azure? I am referring [this method](https://developer.hashicorp.com/vault/docs/platform/k8s/injector/examples#environment-variable-example) for creating credentials from vault into a file.

**Comment by VickyWinner** [comment]
does velero looks for environment variables `AZURE_CLIENT_ID` and `AZURE_CLIENT_SECRET` in velero container to be available so that it can authenticate to AKS?

**Reply by anshulahuja98** [reply]
> does velero looks for environment variables `AZURE_CLIENT_ID` and `AZURE_CLIENT_SECRET` in velero container to be available so that it can authenticate to AKS?

Yes that is correct. In addition you also might need to pass: AZURE_TENANT_ID, AZURE_RESOURCE_GROUP, AZURE_CLOUD_NAME

**Reply by anshulahuja98** [reply]
It is used to authenticate to the storage account(BSL - BackupStorageLocation) and if you are using cloud plugin for snapshotting - it is also used for that purpose.
